16.py

```diff
@@ -0,0 +1,73 @@
+class Vehicle:
+    def __init__(self, number, max_speed):
+        self.number = number
+        self.max_speed = max_speed
+
+    def __str__(self):
+        return f"Vehicle {self.number}, Max Speed: {self.max_speed} km/h"
+
+
+class Bus(Vehicle):
+    def __init__(self, number, max_speed, monthly_fuel_comsumption):
+        super().__init__(number, max_speed)
+        self.monthly_fuel_consumption = monthly_fuel_comsumption
+
+    def __str__(self):
+        return f"Bus {self.number}, Max Speed: {self.max_speed} km/h, Monthly Fuel Consumption: {self.monthly_fuel_consumption} liters"
+
+
+class Tram(Vehicle):
+    def __init__(self, number, max_speed, num_wagons):
+        super().__init__(number, max_speed)
+        if num_wagons > 0 & num_wagons < 4:
```

Comment:
lepiej użyć operatora `and` który służy do logiki w python
operatory `&` i `|` są bardziej do operacji na liczbach binarnych